cEP-0014.md

```diff
@@ -351,3 +351,409 @@ A possible set of deault capabilties can be:
 4. The filtering and selection of bears take place as described in sections above.
 5. After the bears are selected for every language present in the project, coala-quickstart tries automatically to fill the setting values for the bears, failing which, the bear is discarded. In case some inconsistency is encountered here, an attempt is made to resolve it based on scope, failing which, the bear may be discarded, or any one of the possible value is filled in with a warning comment.
 6. The per-language sections are generated, and final `.coafile` is written.
+
+
+## Code Samples
+
+### Collecting extracted information from project
+```py
+from coala_quickstart.info_extractors.GruntfileInfoExtractor import (
+    GruntfileInfoExtractor)
+from coala_quickstart.info_extractors.EditorconfigInfoExtractor import (
+    EditorconfigInfoExtractor)
+from coala_quickstart.info_extractors.PackageJSONInfoExtractor import (
+    PackageJSONInfoExtractor)
+from coala_quickstart.info_extractors.PackageJSONInfoExtractor import (
+    GemfileInfoExtractor)
+
+
+def collect_info(project_dir):
+    gruntfile_info = GruntfileInfoExtractor(
+        ["Gruntfile.js"], project_dir).extract_information()
+
+    editorconfig_info = EditorconfigInfoExtractor(
+        [".editorconfig"], project_dir).extract_information()
+
+    package_json_info = PackageJSONInfoExtractor(
+        ["package.json"], project_dir).extract_information()
+
+    gemfile_info = GemfileInfoExtractor(
+        ["Gemfile"], project_dir).extract_information()
+
+    extracted_info = aggregate_info(
+        gruntfile_info, editorconfig_info, package_json_info, gemfile_info)
+
+    return extracted_info
+
+
+def aggregate_info(*args):
+    """
+    Aggregates inforamtion extracted from multiple ``InfoExtractor``
+    instances to one object.
+    """
+    result = {}
+    for arg in args:
+        # the fname key level can be removed from the current implementation
+        # of the way information is stored as file name is already stored
+        # in source attribute of ``Info`` classes.
+        for fname, extracted_info in arg.items():
+            for info_name, info_instances in extracted_info.items():
+                if result.get(info_name):
+                    result[info_name] += info_instances
+                else:
+                    result[info_name] = info_instances
+    return result
+```
+
+### Collecting all the meta-data
+```py
+from collections import defaultdict
+
+from pyprint.NullPrinter import NullPrinter
+
+from coalib.settings.ConfigurationGathering import get_filtered_bears
+from coalib.output.printers.LogPrinter import LogPrinter
+from coalib.misc.DictUtilities import inverse_dicts
+
+
+used_languages = ["languages", "used", "in", "project"]
+
+
+def get_all_bears(languages):
+    """
+    :param languages: list of languages.
```

Comment:
Add a line explaining what this function does. We do it for even the most trivial functions so that the parser will know what this is about.